src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -205,6 +210,14 @@ static Object preProcessValue(Object value, Schema schema, Schema newSchema) {
         ArrayList<Struct> mapStructs = new ArrayList<>();
         Map<?, ?> map = (Map<?, ?>) value;
         Schema newValueSchema = newSchema.valueSchema();
+        if (keySchema.type() == Schema.Type.STRING) {
+          Map<Object, Object> toReturn = new HashMap<>();
```

Comment:
I'd use a slightly better name. Since the returned object below mentions what it is (`mapStructs`) you can do the same here. Suggestions: `rawMap`, `plainMap`, `processedMap` ... etc